src/EventLoop/Internal/AbstractDriver.php

```diff
@@ -418,7 +418,8 @@ private function invokeMicrotasks(): void
             [$callback, $args] = $this->microtaskQueue->dequeue();
 
             try {
-                $callback(...$args);
+                // Clear $args to allow garbage collection
+                $callback(...$args, ...($args = []));
```

Comment:
So how does this work? You first pass all the elements of the `$args` array into the argument list for `$callback`, and then you overwrite  `$args` with an empty array and also "push" that to the same argument list?

Reply:
Yes, it overwrites the `$args` array (and pushes zero arguments to the argument list) after pushing the original arguments and before invoking the callback.

Reply:
Near trick, it's a tad confusing when you first read the line tbh.

Reply:
A better documentation what happens here would be good especially for people like me that doesn't see what's happening at the first glance and why it's needed.